cmd/mdbd/main.go

```diff
@@ -70,12 +82,36 @@ func getLogger() *log.Logger {
 	return log.New(os.Stderr, "", log.LstdFlags)
 }
 
+// The generator interface generates an mdb from some source.
+type generator interface {
+	Generate(datacentre string, logger *log.Logger) (*mdb.Mdb, error)
+}
+
+// source implements the generator interface and generates an mdb from
+// either a flat file or a url.
 type source struct {
+	// The function parses the data from url or flat file.
 	driverFunc driverFunc
-	url        string
+	// the url or path of the flat file
+	url string
+}
+
+func (s source) Generate(
+	datacentre string, logger *log.Logger) (*mdb.Mdb, error) {
+	return loadMdb(s.driverFunc, s.url, datacentre, logger)
 }
 
```

Comment:
I think it might be cleaner to move these types and their methods into a separate file. Perhaps generators.go?